packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedFieldMessageEmbed.ts

```diff
@@ -0,0 +1,167 @@
+	 * The way in which the items of the object array will be represented in the embed field to be paginated.
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginationFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 *
+	 * new PaginationFieldMessageEmbed()
+	 *    .setTitleField('Test field')
+	 *    .setTemplate({ embed })
+	 *    .setItems([
+	 *       { title: 'Sapphire Framework', value: 'discord.js Framework' },
+	 *       { title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+	 *       { title: 'Sapphire Framework 3', value: 'Discord.js Framework 3' }
+	 *     ])
+	 *    .formatField((item) => `${item.title}\n${item.value}`)
+	 *    .make()
+	 *    .run(message);
+	 * ```
+	 * @param value Item information
+	 */
+	public formatField(value: (value: T, index: number, array: T[]) => any) {
+		this.items = this.items.map(value);
+		return this;
+	}
+
+	/**
+	 * Build the item pagination of the configured array.
+	 * It is required to put the `.make()` and `.run()` method at the end in order to start.
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginationFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 *
+	 * new PaginationFieldMessageEmbed()
+	 *    .setTitleField('Test field')
+	 *    .setTemplate({ embed })
+	 *    .setItems([
+	 *       { title: 'Sapphire Framework', value: 'discord.js Framework' },
+	 *       { title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+	 *       { title: 'Sapphire Framework 3', value: 'Discord.js Framework 3' }
+	 *     ])
+	 *    .formatField((item) => `${item.title}\n${item.value}`)
+	 *    .make()
+	 *    .run(message);
+	 * ```
+	 */
+	public make() {
+		if (!this.fieldTitle.length) throw new Error('The title of the field to format must have a value.');
+		if (!this.items.length) throw new Error('The array introduced has no items.');
+		if (!this.items[0]) throw new Error('The format of the array items is incorrect.');
+
+		this.totalPages = Math.ceil(this.items.length / this.itemsPerPage);
+		this.generatePages();
+		return this;
+	}
+
+	private generatePages() {
+		const parsedTemplate = this.embedTemplate instanceof MessageEmbed ? (this.embedTemplate.toJSON() as MessageEmbedOptions) : this.embedTemplate;
+		for (let i = 0; i < this.totalPages; i++) {
+			const clonedTemplate = new MessageEmbed(parsedTemplate);
+			const fieldsClone = this.embedTemplate.fields;
+			clonedTemplate.fields = [];
+
+			if (!clonedTemplate.footer) clonedTemplate.setFooter(`${i + 1}/${this.totalPages}`);
+			if (!clonedTemplate.color) clonedTemplate.setColor('RANDOM');
+
+			const data = this.paginateArray(this.items, i + 1, this.itemsPerPage);
+			this.addPage({
+				embeds: [clonedTemplate.addField(this.fieldTitle, data.join('\n'), false).addFields(fieldsClone)]
+			});
+		}
+	}
+
+	private paginateArray(items: T[], currentPage: number, perPageItems: number) {
+		const page = currentPage;
+		const perPage = perPageItems;
+		const offset = (page - 1) * perPage;
```

Comment:
Instead of adding one to the page number in the method call and then subtracting one here, would it not be more efficient to remove both arithmetic operations in favour of just passing the actual index?